### PR TITLE
Framegraph: report data as structured protobuf

### PR DESCRIPTION
**Pull request by hevrard**
This change refactors the representation of the Framegraph details
from Go types into protobuf messages. As a result:

- there is a new `-json` flag in the gapit framegraph command to
  export the framegraph data as JSON.

- the formatting into a DOT file now happens in gapit.

Bug: b/172853011

### CODE REVIEW

**Review by johnplate — APPROVED** [review]
LGTM, but wait for another approval

**Comment by hevrard** [comment]
Notes from discussion with Pascal:

- [ ]  merge the loadOp / storeOp enums into one

**Review by pmuetschard — COMMENTED** [review]
LGTM, but fix the presubmit... we (unfortunately) are using clang-format to format protos, so it expects c++ comments - two spaces before `//`.

**Pull request merged**
